src/legends.js

```diff
@@ -0,0 +1,30 @@
+import {exposeScale, Scale} from "./scales.js";
+import {legendColor} from "./legends/color.js";
+
+export function legend({color, ...options}) {
+  if (color != null) {
+    return legendColor(exposeScale(Scale("color", undefined, color)), {
+//    ...color,
+    legend: color.legend,
+    label: color.label,
+    tickSize: color.tickSize,
+    width: color.width, 
+    height: color.height,
+    marginTop: color.marginTop,
+    marginRight: color.marginRight,
+    marginBottom: color.marginBottom,
+    marginLeft: color.marginLeft,
+    ticks: color.ticks,
+    tickFormat: color.tickFormat,
+    tickValues: color.tickValues,
+    format: color.format,
+    className: color.className,
+    swatchSize: color.swatchSize,
+    swatchWidth: color.swatchWidth,
+    swatchHeight: color.swatchHeight,
+    columns: color.columns,
```

Comment:
Why are these options also allowed under *color*? Like, I think we want you to say

```js
Plot.legend({color: …, width: 640})
```

rather than 

```js
Plot.legend({color: {…, width: 640}})
```

Reply:
OK, the confusion was coming from my unit tests, where I had a mix of scale options and legend options. I've cleaned this. I've left _ticks_, _tickFormat_, _tickValues_ and _format_ in comments as it would feel consistent to have them as scale options (as they already are for positional scales). But it's not necessary.